components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -179,13 +172,11 @@ EthSignTypedDataHelper::EncodeData(const std::string& primary_type_name,
       if (!encoded_field) {
         return std::nullopt;
       }
-      result.insert(result.end(), encoded_field->begin(), encoded_field->end());
+      base::Extend(result, *encoded_field);
```

Comment:
I'm a bit confused about this loop. `EncodeField` returns `<EthSignTypedDataHelper::Eip712HashArray`, which is a `array<uint8_t, 32u>`. So it is extending the vector with this value, but the loop doesn't bail out at this point, so potentially there could be another call, that could extend the vector further. Maybe this doesn't happen because of the data being passed in. Am I missing something?

Reply:
We loop through schema of `primary_type` and for each item encode corresponding value from `data`. This `result` array is expected to be extended many times.

Reply:
Sorry, I'm a bit confused now. Each time we get here we extend this array 32 bytes (I think). But then we return as:

```
return std::make_pair(result, std::move(sanitized_data));
```

Wouldn't this cause any data added to the vector past the first call to be dropped? I feel I'm missing something here, but the vector is being passed as the value for the `EthSignTypedDataHelper::Eip712HashArray` entry in the return pair.

Reply:
Np, mate :)

After `1 + N` calls of `base::Extend` `result` would become in typical case:
`type-hash | encoded-field-1 | encoded-field-2 | ... | encoded-field-N`
which is `(1 + N) * 32` bytes

Then we return `result` along with sanitized data

Reply:
I was misunderstanding the return type.